### ISSUE TITLE
Decoded Vectored Exception Handlers on x64 Windows 10 point to invalid memory addresses

### ISSUE THREAD

**Issue by olliencc** [x64dbg/x64dbg#2814]
### Operating System

Windows 10 64bit Build 19042

### x64dbg Version

Snapshot January 1 2022

### Describe the issue

When using the exhandlers command I get a set of addresses which are not valid memory.

### Steps to reproduce

1.  Compile the attached 
2.  Run it
3.  Attach x64dbg
4.  Load NTDLL symbols
5.  execute the exhandlers command

### Attachments

[MEMGUARD.zip](https://github.com/x64dbg/x64dbg/files/7798316/MEMGUARD.zip)

**Comment by olliencc** [comment]
OK, I got this working in my own code - if it helps here is the code:
https://gist.github.com/olliencc/de179ea37eeee853a621045a4ff775ff

the structure I ended up using was the below with the encoded point being in VectoredHandler3.

```#pragma pack(2)
struct VEH_HANDLER_ENTRY
{
	LIST_ENTRY Entry;
	PVOID VectoredHandler;
	PVOID VectoredHandler2;
	PVOID VectoredHandler3;
};

**Comment by mrexodia** [comment]
Thanks! I guess they changed the structure at some point? This feature definitely used to work on Windows 8.1 at least…

**Comment by blaquee** [comment]
Related information: https://research.nccgroup.com/2022/01/03/detecting-anomalous-vectored-exception-handlers-on-windows/

**Comment by mrexodia** [comment]
Thanks for pointing this out, I looked into it and I think I found a somewhat generic way to support all versions of Windows. With regards to the anomalous vectored exception handlers, also check out `Wow64PrepareForException` and instrumentation callbacks 🙂

With regards to your code, I couldn't look at it because it's AGPL. Perhaps use a more liberal license, especially considering your sources all use more liberal licenses.
